src/Common/src/TypeSystem/IL/HelperExtensions.cs

```diff
@@ -93,5 +93,117 @@ public static MetadataType GetKnownType(this ModuleDesc module, string @namespac
 
             return type;
         }
+
+        public static MetadataType GetSafeHandleType(this TypeSystemContext context)
+        {
+            return context.SystemModule.GetKnownType("System.Runtime.InteropServices", "SafeHandle");
+        }
+
+        public static MetadataType GetPInvokeMarshal(this TypeSystemContext context)
+        {
+                return context.SystemModule.GetKnownType("System.Runtime.InteropServices", "PInvokeMarshal");
+        }
+
+        /*      
+                TODO: Bring CriticalHandle to CoreLib
+                https://github.com/dotnet/corert/issues/2570
+
+                public static MetadataType GetCriticalHandle(this TypeSystemContext context)
+                {
+                        return context.SystemModule.GetKnownType("System.Runtime.InteropServices", "CriticalHandle");
+                }
+
+
+                TODO: Bring HandleRef to CoreLib
+                https://github.com/dotnet/corert/issues/2570
+
+                public static MetadataType GetHandleRef(this TypeSystemContext context)
+                {
+                    get
+                    {
+                        return context.SystemModule.GetKnownType("System.Runtime.InteropServices", "HandleRef");
+                    }
+                }
+        */
+        public static MetadataType GetStringBuilder(this TypeSystemContext context)
```

Comment:
It would be nice to move these into a separate static class in the `Internal.TypeSystem.Interop` namespace so that it doesn't pollute `TypeSystemContext` with a bunch of extension methods in the more global namespace.